### ISSUE TITLE
[WotLK] NPC's Minions can not be attacked.

### ISSUE THREAD

**Issue by Darth-Postal** [cmangos/issues#1044]
**Current behavior**: 
1. If NPC has a Succubus Minion, she doesn't help her master and also can not be attacked. (Originally reported by @adellaci in https://github.com/unified-db/Database/issues/237)
![wowscrnshot_081416_153904](https://cloud.githubusercontent.com/assets/19946716/17649155/fef170e2-6235-11e6-90a1-db5fe365567d.jpg)
2. If NPC has a Voidwalker Minion, it can't be attacked (but unlike Succubus, helps its master).
![wowscrnshot_081416_152329](https://cloud.githubusercontent.com/assets/19946716/17649159/29fd4766-6236-11e6-8207-2139ca3649c8.jpg)

**Expected behavior**: 
Minions should be attackable and help their masters.

**Steps to reproduce**:
1. Attack an NPC with a minion. Examples above are [Ilkrud Magthrull](http://www.wowhead.com/npc=3664/unused-ilkrud-magthrull) in Ashenvale and [Wastewander Shadow Mage](http://www.wowhead.com/npc=5617/wastewander-shadow-mage) in Tanaris
2. Attempt to kill their minion.

**Client version**: 
3.3.5a

**Commit hash**: 
https://github.com/cmangos/mangos-wotlk/commit/7d882c6060e97e937ea1bae192fd6d08a508e272

**Database version**: 
UDB 521

**Operating system**: 
Win 7 SP1 x64

PS. I can't say about Ilkrud, but I'm positive, those Shadow Mages' Voidwalkers worked fine a few months ago. Possible problem with recent Pet Patch.
PPS. If I spawn the minion via GM commands (or simply kill/respawn the minion with GM commands), the minion attacks and can be attacked, so this is 100% not a DB issue.

**Comment by Grz3s** [comment]
this may also be related to this:
https://github.com/cmangos/issues/issues/1039#issuecomment-238992493
```quote
now ... if your controled creature gets aggro from anyone ...you loosing control on it... he just runs for few secs .. out of your control... ...and this repeats every moment, when you are hit by someone :(
 Basically it looks now like this:
https://youtu.be/kdSEk3BS9SU
```
not only that ... if u want to attack anyone ... it shows "EVADE" on creature every your hit.

**Comment by Phatcat** [comment]
on it

**Comment by Phatcat** [comment]
@Grz3s Unrelated. "somewhat" at least.

Just confirmed as NOT a core issue; both classic and TBC work fine; this seems to be either an ACID (with wrong summon maybe) or DB issue.

classic:
![skaermbillede 61](https://cloud.githubusercontent.com/assets/1907820/17655179/527ef57a-62ab-11e6-920f-41ffe934a6db.png)

TBC:
![skaermbillede 63](https://cloud.githubusercontent.com/assets/1907820/17655184/5c9ed8f4-62ab-11e6-8004-f001c320348d.png)

**Comment by Phatcat** [comment]
If you look at the bottom-left corner of the last screenshot you will also notice that we are currently NOT sending any combat info regarding minions to the player. (The info that it spawns doesn't count as that info comes from the master of the minion.
I didn't notice this when taking the screenshot, I switched to combat info to show you that it does attack me but meh, it isn't even logged. It's however very easy to see the minion is in the middle of a melee attack, though. He did roughly about 14 damage (critted for 28 on a hit) which.. also seems wrong, especially considering I'm in level 1 starter gear. That is also not a core issue, and seems to be exclusive to the TBC branch.

**Comment by Phatcat** [comment]
Furthermore, creature minions don't even use petAI, nor PetHandler, as that is exclusive to player controlled pets. Just go check out the code flow of the AISelector. It came as a shock to me as I had been coding around the possibility that all pets, regardless of owner may be using it but nope. At least not yet.

**Comment by Phatcat** [comment]
Okay, that's weird, after messing about a bit in game attacking a few more mobs the combat messages just stopped altogether. That surely can't be right.

**Comment by Phatcat** [comment]
> Possible problem with recent Pet Patch.

Gonna have to leave an 'unlikely' to that one, given the reasons stated above (except for the very off-topic combat log stuff)

Unassigning myself again as I see no issue with the mob appearing (summoning effect must work fine) but a problem with behaviour (flags and AI - and since they don't use PetAI...)

**Comment by Phatcat** [comment]
(Just to anyone wondering, if it was an issue with the spell effect summoning the minion, then ALL minions summoned by that spell effect would have the same issues on wrath - they don't.)

**Comment by Phatcat** [comment]
![skaermbillede 65](https://cloud.githubusercontent.com/assets/1907820/17655673/55f4cd28-62b1-11e6-8bab-56fed50859a9.png)

Even better; shows me attacking him and him getting ready to attack me, as well as a completely empty combat log (which was by the way set to everything - 'everything' at default settings)

**Comment by kartoffelsup** [comment]
> the combat messages just stopped altogether

I believe this has been happening for quite some time. People have been using the following lua script to get it working again:
`/run local f = CreateFrame("frame",nil, UIParent); f:SetScript("OnUpdate", CombatLogClearEntries);`
If I'm not misunderstanding something here of course. :)

**Comment by Darth-Postal** [comment]
No, I'm afraid it's 100% core issue.

I have just compiled an older version of CMangos-WotLK, specifically https://github.com/cmangos/mangos-wotlk/commit/c529b71edb295b634cb80582516acf763ee38286 (that's just before the Pet Patch) and all works fine with same DB and same ACID.

![wowscrnshot_081516_120349](https://cloud.githubusercontent.com/assets/19946716/17660261/e47ceb82-62e0-11e6-9588-37010dc8beb4.jpg)

**Comment by Phatcat** [comment]
It's not like I want to deny it as a core issue but I have to. Let me repeat:

> Unassigning myself again as I see no issue with the mob appearing (summoning effect must work fine) but a problem with behaviour (flags and AI - and since they don't use PetAI...)

> (Just to anyone wondering, if it was an issue with the spell effect summoning the minion, then ALL minions summoned by that spell effect would have the same issues on wrath - they don't.)

What exactly did my pet patch do to these? I'm dying to know now since you're 100% sure you better have proof.

**Comment by Phatcat** [comment]
Yeah, gonna have to eat my own words there again;

(Just to anyone wondering, if it was an issue with the spell effect summoning the minion, then ALL minions summoned by that spell effect would have the same issues on wrath - they don't.)

> they don't.

Oh yes they do. Other creature summons are also bugged. Reassigning myself and shutting up until I've found out what is going on in the Wrath spell effects.

**Comment by Darth-Postal** [comment]
> What exactly did my pet patch do to these? I'm dying to know now since you're 100% sure you better have proof.

I didn't say I'm 100% sure it's because of your patch. I said that I'm 100% sure that it's a core issue, because earlier version of core works fine with same DB, same ACID, same config, same client, same PC and same man behind keyboard. And since your patch is the biggest update that happened over these 2 months, it looks like proverbial butler who did it.

If it came across like I'm bashing your patch, I'm sorry. That never was my intention.

**Comment by xfurry** [comment]
Poke @cyberium 
This might happen after the recent update that fixed the DK minion spell.
I will try to investigate this in the following days, nobody comes with a solution in the meantime.

**Comment by Phatcat** [comment]
I see another db issue from the second picture, an issue across all of the dbs; do you see it as well?

(Hint: What unit class is voidwalkers supposed to be? - look at the bar beneath his health)

This would also indicate that this call from InitStatsforLevel (which by the way works flawlessly for classic and TBC) doesn't work for wrath:

    // Remove rage bar from pets
    SetMaxPower(POWER_RAGE, 0);

Otherwise it would be these calls from the SpellEffect SummonPet interfering:

    NewSummon->InitLevelupSpellsForLevel();
    NewSummon->InitTalentForLevel();

And I honestly don't see how.

... This is getting on my nerves..

**Comment by Phatcat** [comment]
Ehhh.... wait a second, is that 1 picture showing that it has no power bar, and another screenshot showing it has a filled rage bar..... From the same person.

My head hurts.

**Comment by Phatcat** [comment]
Ahhhh, https://cloud.githubusercontent.com/assets/19946716/17660261/e47ceb82-62e0-11e6-9588-37010dc8beb4.jpg is from an old version. 
thank god, SetMaxPower works fine for wrath.

**Comment by cyberium** [comment]
Remove UNIT_BYTE2_FLAG_SUPPORTABLE from -> https://github.com/cmangos/mangos-wotlk/blob/master/src/game/Pet.cpp#L1943
and it will work again

**Comment by Phatcat** [comment]
@cyberium

With all due respect that flag was never meant to have been set for non player creatures. That it has been has been an error from the very beginning.

Furthermore the places where I've actually added it (all besides this one tiny place) were all places that it was needed in order to be able to heal your own pet, speak with @killerwife for more info.

also that line should NOT be pulled out, but rather given a condition to only run if the pet owner is a player; if we remove this altogether, then mini-pets (critters) that players summon will no longer correctly display their auras, and neither will they be healable/bandageable. try reverting https://github.com/cmangos/mangos-wotlk/commit/08443e6d24530e0746e967eb8f416e38471d24fd and see what happens.

Furthermore, look at the name of the flag before I changed it. Sanctuary. Yeah that doesn't fit at all. What does sanctuary have to do with this? because setting it for mobs will make you not able to attack them?

To sum it up, please speak with me before telling me something has to be pulled back out of the core, let me defend my work; it very well may be that the core is just still rather bugged (which is my standpoint in this scenario) than my changes being wrong, remember to consider that.

**Comment by Phatcat** [comment]
https://github.com/cmangos/mangos-wotlk/commit/32e3eeb4dfe5b27507b81dd49372f2c1f450545e

**Comment by cyberium** [comment]
Well i'll make it short.
- Thank you for the respect. The following message is full of respect about you and your work so dont consider it as an attack.
- Something that dont break things on TBC and Classic doesn't mean that it dont break things on Wotlk. That are different core/client!
- Do you mean i didnt looked enough on what you did and how it worked before?
- To sum it up too i do not need you agreement to do any remark here. I worked on it in hurry this morning before i have to leave so i didnt had time to write more than i did.
- You did an excelent job on this part but as other said this bug was introduced by your work. This is normal, every line we add in this project may introduce new bugs.
- You, 'unassigned yourself" at least 2 times wich mean for me that you needed some help thats why i gave you some hint and said that flags need more work.
Next time you'll have to reevaluate more your understainding and instead to be ready to defend your work with your 'theet ready to bite' (sorry not sure this expression is well translated) please permit yourself to doubt on your own work.

Finaly I assume you are right that this flags is only for players.  Others will certainly confirm it work or not. Its however maybe the same case on others cores. This may need some test because even if others cores seem to work with this flag they may also work without for owner != player. So if its the case sync all core.

Thank you.

**Comment by adellaci** [comment]
It appears to be working correctly to night. 
CMaNGOS/0.18 (2016-08-17 17:02:00 - 58bfd78bff593623bffeb85a60625ce5cccf0cf7) for Linux_x64

**Comment by Phatcat** [comment]
@Darth-Postal @Grz3s You guys satisfied for now with the fix? If so we can close this.

**Comment by adellaci** [comment]
Thank you @Phatcat

**Comment by Grz3s** [comment]
I did a quick look ...and yes minions can be attacked again etc... so ...yes close it...
If anything will come in near future im sure ppl will report it ;)
:+1:

**Comment by Phatcat** [comment]
Thank @cyberium for pointing out my mistake. And thank you for initially reporting the issue on udb.

Was a mistake on all cores that can be traced all the way back to my first pet patch if I am not mistaken, weird it only created trouble for wrath. I won't waste time pondering that now it's been removed, though, may as well move on to some other issues.
